### DISCUSSION TITLE
Anchor links copied from project READMEs now add a `?tab=readme-ov-file` query parameter, making them harder to read

### DISCUSSION THREAD

**Discussion by pirate** [community/community#70577] [Repositories]
### Select Topic Area

Product Feedback

### Body

When copying anchor link to a specific section in a project's README, the URLs now contain an extra query parameter indicating the tab: `?tab=readme-ov-file`, e.g.:

`https://github.com/ArchiveBox/ArchiveBox?tab=readme-ov-file#quickstart`

<img src="https://github.com/community/community/assets/511499/cc25190e-a40e-4c84-bf17-ebbdcd59dd2a" width="400px" alt="screenshot of right-click copying section header URL from a README">

I understand why this is needed for other tabs, but it adds unnecessary clutter and makes links harder to read for the most common default case of linking to a section of the README.

The parameter can be manually deleted from the URL and it still works the same, but this extra cleanup work is something most people wont do. With the current default, we'll end up with tons of links shared around the internet that look like this:

`https://github.com/ArchiveBox/ArchiveBox?tab=readme-ov-file#quickstart`

instead of this:

`https://github.com/ArchiveBox/ArchiveBox#quickstart`

You can already see them starting to spread: https://www.google.com/search?q=%22readme-ov-file%22

If the UI/web team would kindly consider removing this parameter in the default case I think it may improve the link sharing UX for everyone! Github is doing such an awesome job lately with all the new features, I really appreciate the new tabs feature aside from this nit, and thank you to whoever ends up taking a look at this change!

**Comment by vogler** [comment]
It adds `?tab=readme-ov-file` on any click in the README and rewrites history which is just terrible:

https://github.com/community/community/assets/493741/413f9ffa-1205-4d8d-af1a-dac4bb0311d1

**Comment by roke-julian-lockhart** [comment]
> https://github.com/orgs/community/discussions/70577#discussion-5754901

@pirate, I don't see why its inclusion is problematic, because your explanation is rather vague — how is it confusing?

**Reply by pirate** [reply]
Because it makes README URLs unnecessarily longer and harder to understand at a glance when sharing. Just look at the example I gave and try to read both fast and see which one is easier:
<img width="541" alt="image" src="https://github.com/community/community/assets/511499/54bd1052-d3df-4538-959b-9003b336c4f0">

**Reply by roke-julian-lockhart** [reply]
> https://github.com/orgs/community/discussions/70577#discussioncomment-8269223

@pirate, I don't agree. For instance, when even the *name* of the ReadMe file itself has changed, `?tab=readme-ov-file` is better, due to it demonstrating to the user that the user is being focused upon the ReadMe file. This is useful because *then* the fragment (`#`)-delimited section demonstrates that it refers to a heading. However, I agree that ultimately it is flawed — because it's not fragment-delimited, instead an argument, I expect that it's processed server-side. This appears nonsensical to me, because the entire page is loaded regardless. Were it not, it would be a performance improvement too.

Regardless, I've realized that usually, linking to https://github.com/ArchiveBox/ArchiveBox/blob/22aae92e959c4758ea4fc57dcd62ee757de14682/README.md instead is better, because most references to a ReadMe refer to information best correlated to the version relevant to the reference.

**Reply by vogler** [reply]
Come on... This just adds unnecessary noise for 99% of uses (notes, people sharing links to projects). See my video above - it even adds this when you don't link to a heading. Even if, the default tab shouldn't have any parameters, `?tab=coc-ov-file` etc. are fine.

Whether to use permalinks is a different topic and depends on the context.

**Reply by fenn** [reply]
it also breaks uniqueness of the repository URL. now there are two URLs that point to the same place, which can result in duplicate bookmarks, problems with internet archive captures, and so on.

**Reply by silverwind** [reply]
Definitely get rid of that ugly param for the readme tab. It's just polluting the URL with unnecessary info.

**Reply by ioRekz** [reply]
I'm playing with a browser prototype that struggle with that (and probably a lot more things 😅) because github readme -> click link -> browser back button returns to a different URLs which my prototype can't differentiate from a entirely new page. Anecdotical on my case but might surface similar issues

**Comment by vogler** [comment]
Here's a quick fix using Tampermonkey: https://gist.github.com/vogler/74edff6de37c3a13eeff8c99c6bed910
It removes `?tab=readme-ov-file` from the URL and anchor links still work.
If you want `?tab=readme-ov-file` included in the URL, just click on the same anchor link twice.

```js
    const rewrite = () => location.search == '?tab=readme-ov-file' && (console.log('rewrite', location.href) || history.replaceState({}, null, location.href.replace(location.search, '')));
    window.onpopstate = rewrite; // on navigation
    rewrite(); // on initial load
```

**Reply by silverwind** [reply]
Works in Violentmoney as well and I think if you add `.user.js` extension to your gist filename it should be one-click installable from the raw link in all userscript managers.

**Reply by evdcush** [reply]
The amount of Tampermonkey scripts needed to restore basic, prior UX to GitHub is getting silly...

 

Have an issue with the new bigbrain UI/UX changes?
Did GitHub break prior functionality and support and ignore bug reports?

For instance, markup rendering for the native file format of Python documentation (reStructuredText):
- https://github.com/orgs/community/discussions/86715
- https://github.com/orgs/community/discussions/121163


GH: Lol, not Copilot related 🤣

**Reply by phanirithvij** [reply]
There's also refined-github extension, it adds useful improvements to github UX, https://github.com/refined-github/refined-github/issues/6904

**Comment by westurner** [comment]
When sharing GitHub URLs on sites that abbrevi... links, it's really frustrating to not be able to read the url because there's an unnecessary ?tab=readme-ov-file before the URL fragment:

   github.com/org/repo?tab=readme-ov...
   github.com/org/repo#readme-heading

This was an original GitHub feature; clean deeplink urls for README headings

**Comment by didvc** [comment]
I use a highlighter browser extension which heavily depends on URL and params, which is the problematic part at least for me.

**Comment by ryuheechul** [comment]
I'm curious... what does `ov` (in `readme-ov-file`) even stand for? Anyone knows?

**Reply by pirate** [reply]
I'd guess "open view"

**Reply by xpeuvr327** [reply]
maybe "overview"

**Reply by nickolasjadams** [reply]
I think overview is likely it. Assuming the idea is that it's the first readme tab, and that you can add more files and they'll show up in the tabs. But have they documented what those filenames should even be anywhere? Is this a half-baked feature wip?

**Reply by nickolasjadams** [reply]
<img width="663" height="64" alt="image" src="https://github.com/user-attachments/assets/dff87eb1-6f17-4c0e-bfb1-d40df2c0d0fd" />

Found some more information here.

https://github.com/orgs/community/discussions/86658#discussioncomment-9116451

and here.

https://gist.github.com/jakebrinkmann/c63eaedbe384516e4a7bc133c1e1066b

I added `touch CHANGELOG.md CONTRIBUTING.md SUPPORT.md LICENSE.md CONTRIBUTORS.md AUTHORS.
md ACKNOWLEDGEMENTS.md CODE_OF_CONDUCT.md SECURITY.md SUPPORT.md GOVERNANCE.md TESTING.md`

The screenshot shows what appeared in the tabs of my repo. 

So README.md, CODE_OF_CONDUCT.md, CONTRIBUTING.md, LICENSE.md, and SECURITY.md are all options for the tab url param.

**Comment by phanirithvij** [comment]
please remove this, this is messing up with bookmarks

**Comment by woj-tek** [comment]
Yet another "+1". It's so annoying when it pop ups into the url... it's just pointless noise in 99% of the cases…

**Comment by reorx** [comment]
Join the petition and vote to fix this.

**Comment by ihaettypo** [comment]
What the fuck this is still a thing after  2 years?

**Comment by Lamby777** [comment]
I installed the userscript but this is still obnoxious... Even with the script, I can't copy clean links from README file headings, because GitHub inserts this stupid querystring param into the link URL every time you try to copy a heading link or click on it to update your address bar URL. I looked through the README in question-- the file didn't contain `readme-ov-file` anywhere.

Edit: 3 weeks later and this once again caused me annoyance on a different device that doesn't have the userscript installed-- almost double-bookmarked a page because of this stupid bug.

**Comment by evdcush** [comment]
Users might be able to work around this behavior with userscripts or manual cleanup, but that’s not really the issue now.

The greater issue is that links with the `ov-file` parameter now **fragment what should be a single, stable reference**.
A single `README` section can appear as false positives for a unique or “different” URLs in history, analytics, or documentation.
Simply because of a query string that adds no real value.

It's a pattern here with github's UI/UX design team (the latest solution in search of a problem) to make product changes for issues that plainly does not exist.

It adds arbitrary surface complexity to the user experience with no observed benefit.

(It's also not clear what the actual "user pain point" was 🤔)

**Comment by doctorcolossus** [comment]
It's not surprising that since Microsoft bought GitHub, they'd continue their age-old policy of forcing unwanted top-down decisions on users. I  expect that they'll continue "improving" GitHub until they elevate it to the same level they did with Skype.

**Comment by westurner** [comment]
Perhaps most frustrating about this issue is that most of these commenters
could've fixed this bug ourselves.

Are there trackbacks from PRs to Discussions, like the original GitHub had
between PRs and Issues and vice-versa?

**Comment by evdcush** [comment]
https://github.com/orgs/community/discussions/70577#discussioncomment-14783187

> Users might be able to work around this behavior with userscripts or manual cleanup, but that’s not really the issue now.
> 
> The greater issue is that links with the `ov-file` parameter now **fragment what should be a single, stable reference**. A single `README` section can appear as false positives for a unique or “different” URLs in history, analytics, or documentation. Simply because of a query string that adds no real value.
> 
> It's a pattern here with github's UI/UX design team (the latest solution in search of a problem) to make product changes for issues that plainly does not exist.
> 
> It adds arbitrary surface complexity to the user experience with no observed benefit.
> 
> (It's also not clear what the actual "user pain point" was 🤔)


# 😎
<img src="https://github.com/user-attachments/assets/f80e32f4-a3e6-46d6-86ca-36f71e66ef63" />



why was was this "feature" added again? what was the "user story"?

"𝙖𝙨 𝙖 𝙪𝙨𝙚𝙧, 𝙄 𝙤𝙛𝙩𝙚𝙣 𝙧𝙚𝙖𝙙 𝙘𝙤𝙙𝙚 𝙤𝙛 𝙘𝙤𝙣𝙙𝙪𝙘𝙩 𝙖𝙣𝙙 𝙡𝙞𝙘𝙚𝙣𝙨𝙚 𝙛𝙞𝙡𝙚𝙨.𝙬𝙝𝙞𝙡𝙚 𝙞'𝙢 𝙧𝙚𝙛𝙚𝙧𝙚𝙣𝙘𝙞𝙣𝙜 𝙩𝙝𝙚𝙨𝙚 𝙛𝙞𝙡𝙚𝙨, 𝙬𝙝𝙞𝙘𝙝 𝙞 𝙛𝙧𝙚𝙦𝙪𝙚𝙣𝙩𝙡𝙮 𝙙𝙤, 𝙞𝙩 𝙞𝙨 𝙖 𝙥𝙖𝙞𝙣 𝙩𝙤 𝙝𝙖𝙫𝙚 𝙩𝙤 𝙖𝙘𝙘𝙚𝙨𝙨 𝙩𝙝𝙚 𝙛𝙞𝙡𝙚𝙨, 𝙗𝙚𝙘𝙖𝙪𝙨𝙚 𝙞 𝙨𝙬𝙞𝙩𝙘𝙝 𝙗𝙖𝙘𝙠 𝙖𝙣𝙙 𝙛𝙤𝙧𝙩𝙝 𝙖 𝙡𝙤𝙩 𝙩𝙤 𝙘𝙧𝙤𝙨𝙨-𝙧𝙚𝙛𝙚𝙧𝙚𝙣𝙘𝙚 𝙩𝙝𝙚𝙢 𝙖𝙨 𝙄 𝙧𝙚𝙖𝙙 𝙩𝙝𝙚 𝙍𝙀𝘼𝘿𝙈𝙀. 𝙤𝙪𝙧 𝙐𝙓 𝙙𝙖𝙩𝙖 𝙨𝙞𝙢𝙪𝙡𝙖𝙩𝙞𝙤𝙣𝙨 𝙨𝙝𝙤𝙬𝙨 𝘼 𝙩𝙖𝙗𝙗𝙚𝙙 𝙫𝙞𝙚𝙬 𝙬𝙤𝙪𝙡𝙙 𝙧𝙚𝙙𝙪𝙘𝙚 𝙩𝙝𝙞𝙨 𝙩𝙝𝙧𝙖𝙨𝙝𝙞𝙣𝙜 𝙗𝙚𝙩𝙬𝙚𝙚𝙣 𝙛𝙞𝙡𝙚𝙨 𝙩𝙝𝙖𝙩 𝙝𝙖𝙨 𝙪𝙣𝙞𝙛𝙤𝙧𝙢𝙡𝙮 𝙙𝙞𝙨𝙩𝙧𝙞𝙗𝙪𝙩𝙚𝙙 𝙫𝙞𝙚𝙬 𝙥𝙖𝙩𝙩𝙚𝙧𝙣𝙨 𝙖𝙣𝙙 𝙚𝙭𝙩𝙚𝙧𝙣𝙖𝙡 𝙡𝙞𝙣𝙠𝙞𝙣𝙜 𝙩𝙧𝙖𝙛𝙛𝙞𝙘."
